src/urllib3/response.py

```diff
@@ -220,6 +221,16 @@ def get_redirect_location(self) -> typing.Union[str, "Literal[False]"]:
     def data(self) -> bytes:
         raise NotImplementedError()
 
+    def json(self) -> Any:
+        """
+        Parses the body of the HTTP response as UTF-8, UTF-16, or UTF-32 encoded JSON using the standard library :func:`json.loads` function.
```

Comment:
You're right. But we can't do this anyway, as the `encoding` parameter of `json.loads` was removed in Python 3.9. However, since we could support additional parsers in the future, including parsers that only support UTF-8 such as `orjson`, I would avoid documenting this at all.

```suggestion
        Parses the body of the HTTP response as JSON.
```

Reply:
I propose to enforce UTF-8, to follow the latest RFC and because I don't like "surprises" like my client accepting UTF-16/UTF-32 when it is not expected. Your point on orjson is also good, and I think enforcing UTF-8 would make it easier to use alternative JSON decoders consistently.

So I would do

```py
data = self.data.decode('utf-8')
return _json.loads(data)
```

In the docstring I would mention that the function can raise `UnicodeDecodeError` or `json.JSONDecodeError`.

orjson could skip the decode step.

Reply:
OK, that sounds good to me!